### ISSUE TITLE
Semantic Segmentation Differences

### ISSUE THREAD

**Issue by Nadavc220** [zhmiao/OpenCompoundDomainAdaptation-OCDA#4]
Hi, 
First of all thanks for this repository. The approach in this article is very interesting.
I have a few questions about the implementation:

1) The training session in main.py consists of training MANN net without scheduling before training the Disentangle Domain Factor Net. Is there any reason to train it before the Disentangle Domain Factor Net? is there any reason to train the MANN network without scheduling prior to training it with scheduling?

2) What were the architectures used in order to train the domain adaptation task on semantic segmentation data. in particular:
- What layer of the DeepLab (I assume V3+) network was used as the last encoder layer for the DA task?
- What was the architecture of the decoder/domain encoder for the reconstruction loss used in the domain disentanglement train?

Thanks.

**Comment by XingangPan** [comment]
@Nadavc220 For question2, we follow the architecture of https://github.com/wasidennis/AdaptSegNet/blob/master/model/deeplab_vgg.py  And added batchnorm to the architecture. The feature before the classifier is used for class memory bank.
As described in the paper, for the semantic segmentation part we did not explicitly distinguish the multiple domains with domain encoder. Instead, we use curriculum learning with probability confidence and the dynamic transferable embedding to ease the learning. So we believe there is still room for improvement in semantic segmentation.
